### ISSUE TITLE
Option to block mxid patterns at registration

### ISSUE THREAD

**Issue by ara4n** [matrix-org/synapse#5129]
A bit like IRC's badnames feature, to prevent people registering obnoxious accounts.

Could be hooked into the spam scanning interface?

**Comment by turt2live** [comment]
beware the Scunthorpe problem

**Comment by Half-Shot** [comment]
Yup, I think it's wise to look for the clear cut examples. Words rather than strings, as well as unambiguous strings, with the expectation that admins would have to snap the occasional person that slips through.
